backend/capellacollab/sessions/routes.py

```diff
@@ -87,70 +96,66 @@ def get_current_sessions(
-    "/",
+    "/readonly",
     response_model=AdvancedSessionResponse,
 )
 def request_session(
-    body: PostSessionRequest,
+    body: PostReadonlySessionRequest,
     db_user: DatabaseUser = Depends(get_own_user),
     operator: Operator = Depends(get_operator),
     db: Session = Depends(get_db),
     token=Depends(JWTBearer()),
 ):
-    assert body.type == WorkspaceType.READONLY
-
     rdp_password = generate_password(length=64)
 
-    owner = db_user.username
+    owner = db_user.name
 
     log.info("Starting persistent session creation for user %s", owner)
 
     existing_user_sessions = database.get_sessions_for_user(db, owner)
 
-    if body.repository in [
-        session.repository for session in existing_user_sessions
-    ]:
+    project = get_project_by_slug(db, body.project_slug)
+    if not project:
         raise HTTPException(
-            status_code=404,
-            detail={
-                "err_code": "existing_session",
-                "reason": f"You already have a open Read-Only Session for the repository {body.repository}. Please navigate to 'Active Sessions' to Reconnect",
+            404,
+            {
+                "reason": f"The project with name {body.project_slug} was not found.",
+                "technical": f"No project {body.project_slug} found.",
             },
         )
-    verify_project_role(repository=body.repository, token=token, db=db)
-    git_model = git_models_crud.get_primary_model_of_repository(
-        db, body.repository
-    )
-    if not git_model:
+
+    verify_project_role(project.name, token, db)
+
+    model = get_model_by_slug(db, body.project_slug, body.model_slug)
+    if not model:
         raise HTTPException(
-            status_code=404,
-            detail={
-                "err_code": "git_model_not_found",
-                "reason": "The Model has no connected Git Model. Please contact a project manager or admininistrator",
+            404,
+            {
+                "reason": f"The model with name {body.model_slug} was not found.",
+                "technical": f"No model {body.model_slug} found.",
             },
         )
 
-    revision = body.branch or git_model.revision
-    if body.depth == DepthType.LatestCommit:
-        depth = 1
-    elif body.depth == DepthType.CompleteHistory:
-        depth = 0
-    else:
+    models = [
+        m
+        for m in project.models
+        if m.git_models and m.version.id == model.version.id
+    ]
+    if not models:
         raise HTTPException(
-            status_code=400,
+            status_code=404,
             detail={
-                "err_code": "wrong_depth_format",
-                "reason": f"Depth type {depth} is not allowed.",
+                "err_code": "git_model_not_found",
+                "reason": "The Project has no connected Git Model. Please contact a project manager or admininistrator",
             },
         )
+
+    docker_image = get_readonly_image_for_version(model.version)
```

Comment:
When the `version.tool` has no `readonly_docker_image_template`, this call raises an error that should be handled (or even better imo, the button should not be availble when there is no `readonly_docker_image_template`).

Reply:
For now, the backend will return a 409 (Conflict) to address the missing read-only docker image.